### PR TITLE
Feature/119 juxstapose feature to image

### PR DESCRIPTION
**Pull request by laurentiu-ilici**

### CODE REVIEW

File: src/plugins/image/display/image-display.js [hunk 1]
```diff
@@ -1,48 +1,68 @@
...
+);
+
+
+const revealEffect = (content, clientConfig) => {
+  const { effect } = content;
+  return (
+    <ReactCompareSlider
+      position={effect.startPosition}
+      portrait={effect.orientation === ORIENTATION.vertical}
+      itemOne={<ReactCompareSliderImage src={getSource(content.sourceType, content.sourceUrl, clientConfig.cdnRootUrl)} alt="Image one" />}
```
Comment (ahelmberger) [inline src/plugins/image/display/image-display.js:42]: The `alt` attribute should probably not be hard-coded

File: src/plugins/image/display/image-display.js [hunk 2]
```diff
@@ -1,48 +1,68 @@
...
+      itemTwo={<ReactCompareSliderImage src={getSource(effect.sourceType, effect.sourceUrl, clientConfig.cdnRootUrl)} alt="Image two" />}
+    />
```
Comment (ahelmberger) [inline src/plugins/image/display/image-display.js:44]: We also need to show somehow the copyright info for the reveal type of images

File: src/plugins/image/editing/image-editor.js [hunk 3]
```diff
@@ -69,24 +69,67 @@ class ImageEditor extends React.Component {
...
+            type: EFFECT_TYPE.hover
+          }
+        });
+        break;
+      case EFFECT_TYPE.none:
+        this.changeContent({ effect: null });
+        break;
+      default:
+        this.changeContent({ effect: null });
+
```
Comment (ahelmberger) [inline src/plugins/image/editing/image-editor.js:111]: Did ESLint not require a `break` in this default case? :thinking:

File: src/plugins/video/video.yml [hunk 4]
```diff
@@ -7,6 +7,9 @@ source:
 externalLink:
   en: External link
   de: Externer Link
+internalLink:
```
Comment (ahelmberger) [inline src/plugins/video/video.yml:10]: Actually, we do not need this. I realized the translation key should be "elmuCdn", not "internalLink" (that was a bug!)

File: src/plugins/image/editing/image-editor.js [hunk 5]
```diff
@@ -69,24 +69,67 @@ class ImageEditor extends React.Component {
...
+    effect.startPosition = Math.max(0, Math.min(100, newPosition));
+    this.changeContent({ effect });
+  }
+
   render() {
     const { docKey, content, clientConfig, t } = this.props;
-    const { type, url, maxWidth, text, hover } = content;
+    const { sourceType, sourceUrl, maxWidth, text, effect } = content;
+
+    const effectType = (effect && effect.type) || EFFECT_TYPE.none;
```
Comment (ahelmberger) [inline src/plugins/image/editing/image-editor.js:132]: could be: `const effectType = effect?.type || EFFECT_TYPE.none;` (what do you think about this coding style?)

File: src/plugins/image/display/image-display.js [hunk 6]
```diff
@@ -1,48 +1,68 @@
...
+      className={`Image-img u-max-width-${content.maxWidth || 100}`}
+      src={getSource(content.effect.sourceType, content.effect.sourceUrl, clientConfig.cdnRootUrl)}
+    />
+    <div
+      className="Image-copyrightInfo"
+      dangerouslySetInnerHTML={{ __html: githubFlavoredMarkdown.render(content.effect.text || '') }}
+    />
+  </div>
+);
+
```
Comment (adrianaluchian) [inline src/plugins/image/display/image-display.js:34]: I'm adding the linting rule for empty lines in my PR, fear not.

Comment (laurentiu-ilici) [reply src/plugins/image/display/image-display.js:34]: Ah I have 2, ups.

File: src/plugins/image/display/image-display.js (hunk 1 above)
Comment (laurentiu-ilici) [reply src/plugins/image/display/image-display.js:42]: hmm, didn't think of those

File: src/plugins/image/editing/image-editor.js [hunk 7]
```diff
@@ -20,39 +21,38 @@ class ImageEditor extends React.Component {
...
   }
 
-  handleHoverExternalUrlValueChanged(event) {
+  handleEffectExternalSourceUrlValueChanged(event) {
     const { value } = event.target;
-    const hover = { ...this.props.content.hover };
-    hover.url = value;
-    this.changeContent({ hover });
+    const effect = { ...this.props.content.effect };
+    effect.sourceUrl = value;
```
[the remark this answers is not in the document]
Comment (laurentiu-ilici) [reply src/plugins/image/editing/image-editor.js:32]: True, done

File: src/plugins/image/editing/image-editor.js (hunk 3 above)
Comment (laurentiu-ilici) [reply src/plugins/image/editing/image-editor.js:111]: nope

File: src/plugins/image/editing/image-editor.js (hunk 5 above)
Comment (laurentiu-ilici) [reply src/plugins/image/editing/image-editor.js:132]: I thought it comes from typescript and I was dying inside to write this...

File: src/plugins/image/display/image-display.js (hunk 1 above)
Comment (ahelmberger) [reply src/plugins/image/display/image-display.js:42]: We could use the content of the `text` property

File: src/plugins/image/display/image-display.js [hunk 8]
```diff
@@ -1,48 +1,68 @@
...
+      position={effect.startPosition}
+      portrait={effect.orientation === ORIENTATION.vertical}
+      itemOne={<ReactCompareSliderImage src={getSource(content.sourceType, content.sourceUrl, clientConfig.cdnRootUrl)} alt="Image one" />}
+      itemTwo={<ReactCompareSliderImage src={getSource(effect.sourceType, effect.sourceUrl, clientConfig.cdnRootUrl)} alt="Image two" />}
+    />
   );
+};
+
+function ImageDisplay({ content, clientConfig, githubFlavoredMarkdown }) {
+  if (content.effect && content.effect.type === EFFECT_TYPE.reveal) {
```
[the remark this answers is not in the document]
Comment (ahelmberger) [reply src/plugins/image/display/image-display.js:49]: It's a JS feature, too.

File: src/plugins/image/display/image-display.js [hunk 9]
```diff
@@ -1,48 +1,68 @@
...
-      <img
-        className={`Image-img u-max-width-${content.maxWidth || 100}`}
-        src={getSource(content.hover.type, content.hover.url, clientConfig.cdnRootUrl)}
-        />
-      <div
-        className="Image-copyrightInfo"
-        dangerouslySetInnerHTML={{ __html: githubFlavoredMarkdown.render(content.hover.text || '') }}
-        />
-    </div>
+const hoverEffect = (content, githubFlavoredMarkdown, clientConfig) => (
```
[the remark this answers is not in the document]
Comment (laurentiu-ilici) [reply src/plugins/image/display/image-display.js:22]: Can do, this was just the old code a bit copy pasted :D

File: src/plugins/video/video.yml (hunk 4 above)
Comment (laurentiu-ilici) [reply src/plugins/video/video.yml:10]: Fixed

File: src/plugins/image/display/image-display.js (hunk 2 above)
Comment (laurentiu-ilici) [reply src/plugins/image/display/image-display.js:44]: Shown

File: src/plugins/image/display/image-display.js (hunk 1 above)
Comment (laurentiu-ilici) [reply src/plugins/image/display/image-display.js:42]: removed

File: src/plugins/image/display/image-display.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (laurentiu-ilici) [reply src/plugins/image/display/image-display.js:22]: Done

File: src/plugins/image/editing/image-editor.js (hunk 5 above)
Comment (ahelmberger) [reply src/plugins/image/editing/image-editor.js]: :smile:

**Pull request merged**